src/api/resources/index.ts

```diff
@@ -149,7 +152,19 @@ export class ResourcesApi {
       }
     }
     if (provider) {
-      return this.resProvider[resType]?.get(provider)?.setResource(resId, data)
+      try {
+        const r = await this.resProvider[resType]
+          ?.get(provider)
+          ?.setResource(resId, data)
+        this.app.handleMessage(
+          provider as string,
+          this.buildDeltaMsg(resType, resId, data),
+          SKVersion.v2
+        )
+        return r
+      } catch (e) {
+        return Promise.reject(new Error(`Error deleting ${resType} ${resId}`))
```

Comment:
The error message is _deleting_ but this is setResource.

PS In this particular case I think  `...then((r) => { this.app.handleMessage(...); return r }` would have communicated the intent pretty clearly.

Reply:
Fixed error message.
Use .`then()` in place of `try` & `await`.